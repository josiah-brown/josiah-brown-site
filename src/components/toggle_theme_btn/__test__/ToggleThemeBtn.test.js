import {
  render,
  cleanup,
  fireEvent,
  getByTestId,
  getByRole,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import ToggleThemeBtn from "../ToggleThemeBtn";

afterEach(cleanup);

test("renders toggle theme button without crashing", () => {
  render(<ToggleThemeBtn />);
});

test("toggles localStorage on click", () => {
  const { getByTestId } = render(<ToggleThemeBtn />);

  fireEvent.click(getByTestId("toggle-theme-btn"));
  expect(window.localStorage.theme).toBe("dark");

  fireEvent.click(getByTestId("toggle-theme-btn"));
  expect(window.localStorage.theme).toBe("light");
});

test("toggles root classlist on click", () => {
  const { getByTestId } = render(<ToggleThemeBtn />);

  fireEvent.click(getByTestId("toggle-theme-btn"));
  expect(window.document.documentElement).toHaveClass("dark");
  expect(window.document.documentElement).not.toHaveClass("light");

  fireEvent.click(getByTestId("toggle-theme-btn"));
  expect(window.document.documentElement).toHaveClass("light");
  expect(window.document.documentElement).not.toHaveClass("dark");
});

test("matches toggle theme button snapshot", () => {
  const tree = renderer.create(<ToggleThemeBtn />).toJSON();
  expect(tree).toMatchSnapshot();
});
