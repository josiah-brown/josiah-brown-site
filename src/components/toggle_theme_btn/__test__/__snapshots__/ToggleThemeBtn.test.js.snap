// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches toggle theme button snapshot 1`] = `
<button
  className="absolute z-50 top-8 border-[1px] border-black dark:border-white w-10 h-5 right-0 md:top-12 "
  data-testid="toggle-theme-btn"
  onClick={[Function]}
  type="button"
>
  <div
    className="relative w-full h-full"
  >
    <div
      className="absolute left-0 dark:left-1/2 top-0 w-1/2 h-full border-white dark:border-neutral-800 border-[1px] bg-black dark:bg-white "
    />
  </div>
</button>
`;
